airflow/providers/microsoft/azure/operators/batch.py

```diff
@@ -181,7 +182,12 @@ def __init__(
     @cached_property
     def hook(self) -> AzureBatchHook:
         """Create and return an AzureBatchHook (cached)."""
-        return self.get_hook()
+        return AzureBatchHook(self.azure_batch_conn_id)
+
+    @deprecated(reason="use `hook` property instead.")
```

Comment:
We deprecate with AirflowProviderDeprecation warning

Reply:
Right, I will update it (and the other PR)

Reply:
@eladkal WDYT about this:
https://github.com/apache/airflow/blob/8b15cbf0543d71c7fcb7c96203b1aab05fb237c5/airflow/providers/microsoft/azure/operators/batch.py#L187
Is it ok? or you prefer the classic warning or the custom decorator

Reply:
The decorator uses 2 as stacklevel by default

Reply:
Not near my laptop but need to check if we have open task on this one. We need to verify providers raise the provider deprecation warning rather than the regular one.

Reply:
> but need to check if we have open task on this one

I already checked it

> We need to verify (pre-commit?) providers raise the provider deprecation warning rather than the regular one.

I agree, I will create a task for that